tests/describe_task_queue.go

```diff
@@ -135,11 +135,51 @@ func (s *DescribeTaskQueueSuite) publishConsumeWorkflowTasksValidateBacklogInfo(
 
 	// call describeTaskQueue to verify if the backlog decreased
 	expectedBacklogCount[enumspb.TASK_QUEUE_TYPE_WORKFLOW] = int64(0)
-	s.validateDescribeTaskQueue(tl, expectedBacklogCount, isEnhancedMode, true)
+	s.validateDescribeTaskQueue(tl, expectedBacklogCount, isEnhancedMode, true, true, workflows)
+}
+
+func (s *DescribeTaskQueueSuite) validateBacklogHeadCreateTime(queueType enumspb.TaskQueueType,
+	backlogHeadCreateTime time.Duration, nullBacklogHeadCreateTime bool) bool {
+	if queueType != enumspb.TASK_QUEUE_TYPE_WORKFLOW {
+		if backlogHeadCreateTime != time.Duration(0) {
+			return false
+		}
+	} else if nullBacklogHeadCreateTime {
+		if backlogHeadCreateTime != time.Duration(0) {
+			return false
+		}
+	} else {
+		if backlogHeadCreateTime == time.Duration(0) {
+			return false
+		}
+	}
+	return true
+}
+
+func (s *DescribeTaskQueueSuite) validateAddDispatchTasksRate(queueType enumspb.TaskQueueType, workflows int,
```

Comment:
Basically checks if the two metrics are 0, when the are supposed to be, and non-zero when they are not (when workflows are added and the queue is of workflow type)